framework/src/Volo.Abp.MongoDB/Volo/Abp/Domain/Repositories/MongoDB/IMongoDbBulkOperationProvider.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Volo.Abp.Domain.Entities;
+using Volo.Abp.Domain.Repositories.MongoDB;
+
+namespace Volo.Abp.MongoDB.Volo.Abp.Domain.Repositories.MongoDB
+{
+    public interface IMongoDbBulkOperationProvider
```

Comment:
hi

The `MongoDbRepository` need to use `SessionHandle`  so we should allow it inject the `IMongoDbContextProvider<TMongoDbContext>`

Like EF Core: ` Task InsertManyAsync<TDbContext, TEntity>`,

Reply:
@maliming Solved at Commit 5a546124

Reply:
hi @enisn 

We can inject `IMonoDbContextProvider<TMongoDbContext>` into the implementation class of `IMonoDbBulkOperationProvider`. This means that the `IMongoDbBulkOperationProvider` interface will to be generic.

```cs
public interface IMongoDbBulkOperationProvider<TMongoDbContext, TEntity>
	where TMongoDbContext : IAbpMongoDbContext
	where TEntity : class, IEntity
{
	Task InsertManyAsync(
	   IMongoDbRepository<TEntity> repository,
	   IEnumerable<TEntity> entities,
	   bool autoSave,
	   CancellationToken cancellationToken
	)

	Task UpdateManyAsync(
		IMongoDbRepository<TEntity> repository,
		IEnumerable<TEntity> entities,
		bool autoSave,
		CancellationToken cancellationToken
	)

	Task DeleteManyAsync(
		IMongoDbRepository<TEntity> repository,
		IEnumerable<TEntity> entities,
		bool autoSave,
		CancellationToken cancellationToken
	)
}
```

Maybe we can also inject ` IMongoDbRepository<TEntity> repository` in the implementation class.
```cs

Task InsertManyAsync(
	IEnumerable<TEntity> entities,
	bool autoSave,
	CancellationToken cancellationToken
)
```

Reply:
Repositories have Transient lifetime as default, If the implementation injects another repository, a new instance of repository will be created. I can't be sure about that. The repository invokes that method already. 

Making interface generic is ok, I agree with that. So each DbContext can has different implementation. 
But injecting just another repository for BulkOperationProvider implementation is confusing me. @maliming  

What do you think @hikalkan ?

Reply:
@maliming no need to that for now.